### ISSUE TITLE
Versioned route matching throws TypeError when "maxV" is not set

### ISSUE THREAD

**Issue by kusor** [restify/node-restify#1380]
# Bug Report

When we have two routes applicable to the same request, one typed and the other versioned and not typed, a TypeError exception is thrown.

## Restify Version

Every one since `v2.8.5`, exactly since commit [Route matching should only prefer later routes if version is greater](https://github.com/restify/node-restify/commit/0a45bd380a45cd7fd62d62d3285e7a707e04bc5d)

## Node.js Version

Any.

## Expected behaviour

Exceptions should not be thrown

## Actual behaviour

```
TypeError: Invalid Version: undefined
    at new SemVer (/Users/pedropc/work/node_modules/restify/node_modules/semver/semver.js:281:11)
    at SemVer.compare (/Users/pedropc/work/node_modules/restify/node_modules/semver/semver.js:348:13)
    at compare (/Users/pedropc/work/node_modules/restify/node_modules/semver/semver.js:571:31)
    at Function.gt (/Users/pedropc/work/node_modules/restify/node_modules/semver/semver.js:600:10)
    at /Users/pedropc/work/node_modules/restify/lib/router.js:497:38
    at Array.forEach (native)
    at Router.find (/Users/pedropc/work/node_modules/restify/lib/router.js:493:24)
    at Server._route (/Users/pedropc/work/node_modules/restify/lib/server.js:753:21)
    at routeAndRun (/Users/pedropc/work/node_modules/restify/lib/server.js:711:14)
    at Server._handle (/Users/pedropc/work/node_modules/restify/lib/server.js:731:9)
```

## Repro case

```
var restify = require('restify');

var server = restify.createServer({
    name: 'myapp',
    version: '1.0.0'
});
server.use(restify.plugins.acceptParser(server.acceptable));
server.use(restify.plugins.queryParser());
server.use(restify.plugins.bodyParser());


server.post({
    path: '/:param_one/path/[0-f]+/:param_two/word',
    contentType: 'application/json'
}, function (req, res, next) {
    req.log.warn(req.headers);
    res.setHeader('Content-Length', '0');
    res.send(204);
    next();
});


server.post({
    path: '/:param_one/path/[0-f]+/:param_two/:param_three'
}, function (req, res, next) {
    req.log.warn(req.headers);
    res.setHeader('Content-Length', '0');
    res.send(204);
    next();
});

server.listen(8080, '127.0.0.1', function () {
    console.log('%s listening at %s', server.name, server.url);
});

var assert = require('assert');
var clients = require('restify-clients');

var client = clients.createJsonClient({
    url: 'http://localhost:8080',
    version: '~1.0'
});

client.post({
    path: '/foo/path/6/bar/word',
    accept: 'application/json',
    contentType: 'application/json'
}, function (err, req, res, obj) {
    assert.ifError(err);
    console.log('Server returned: %j', obj);
});

```

## Cause

The bug problem is into version checking here: https://github.com/restify/node-restify/blob/5.x/lib/router.js#L506-L519.
The first time we hit `semver.gt(v, maxV)`, the value for `maxV` is `undefined`.

## Are you willing and able to fix this?

Yes

**Comment by retrohacker** [comment]
Hey @kusor!

Thank you for opening a thorough bug report and an accompanying PR! I'm tagging this as an unverified bug until a maintainer (probably me :wink:) runs the repro case. :heart:

**Comment by kusor** [comment]
@retrohacker thanks for taking a look!

**Comment by retrohacker** [comment]
Reproduced locally

**Comment by retrohacker** [comment]
Closed by #1381
